src/sinks/clickhouse.rs

```diff
@@ -76,8 +76,13 @@ impl SinkConfig for ClickhouseConfig {
         let tls_settings = TlsSettings::from_options(&self.tls)?;
         let client = HttpClient::new(tls_settings)?;
 
+        let mut config = self.clone();
+        let uri = self.endpoint.parse::<UriSerde>()?;
+        uri.merge_auth_config(&mut config.auth)?;
```

Comment:
It's a bit unfortunate that we need to remember to call this merge method everywhere, but I'm having a hard time coming up with a better modeling given they are separate config fields.

Reply:
Yes it is unfortunate, I also don't have any ideas at the moment.